src/app/client/ui/ClientTaskBoard.tsx

```diff
@@ -25,8 +25,8 @@ export const ClientTaskBoard = ({ updateTask }: { updateTask: (taskId: string, p
   /**
    * This function is responsible for calculating the task count based on the workflowStateId
    */
-  const taskCountForWorkflowStateId = (workflowStateId: string): number => {
-    return tasks.filter((task) => task.workflowStateId === workflowStateId).length
+  const taskCountForWorkflowStateId = (workflowStateId: string): string => {
+    return tasks.filter((task) => task.workflowStateId === workflowStateId).length.toString()
```

Comment:
I agree with your point @foleyatwork. Even though not now, in the near future we may have to implement solutions to tackle the overload and maintain smooth UI. 

Using Map is a good initial approach but I thought there are more cons than the props to it. Creating a Map for tasks is a good approach but the Map needs to store additional key value pair which can potentially increase the overall memory footprint of the app. Also, Keeping the Map in sync with the tasks state can be complex. Whenever a task is added, updated, or removed, the Map must be updated accordingly. Beside this, when initializing the Map for the first time, there will be a one-time cost to iterate over all tasks to populate the map. 

Some of the solutions that I are in my mind are as follows: 
- We can use memoization techniques to cache the results of taskCountForWorkflowStateId and filterTaskWithWorkflowStateId functions by using libraries like `reselect` which is designed for memoizing selectors in Redux. 
- Normalize the state structure using libraries like `normalizr` to store tasks by their IDs and maintain separate lists of task IDs for each workflowStateId. This can make accessing tasks by their workflow state more efficient.
- We can also implement batch processing. Or, web workers to offload heavy computations such as above two functions out of the main thread. 

Let me know your thoughts on this @foleyatwork. 

cc: @sazanrjb @rrojan @arpandhakal

Reply:
Based on your response, I can tell you understand the performance tradeoffs and we can implement some of these optimizations in the future when we start to get heavy usage of the tasks app.

I really like the memoization and the normalizr options. I'm not as into the web workers approach, I think that approach runs into other performance and maintainability issues.

If we upgrade to React 19 and use the React Compiler, we might get some of the memoization for free.

Anyway, we can address these things when the time is right. Thanks!

Reply:
Upgrading to React 19 sounds good to me. Sure, we will address these things in the upcoming days when time is right. Thank you for pointing this out.